rskj-core/src/test/java/co/rsk/peg/BridgeSupportFlyoverTest.java

```diff
@@ -1280,6 +1449,118 @@ public void registerFastBridgeBtcTransaction_funds_sent_to_retiring_fed_after_RS
         Assert.assertEquals(co.rsk.core.Coin.fromBitcoin(valueToSend).asBigInteger(), result);
     }
 
+    @Test
+    public void registerFastBridgeBtcTransaction_sum_of_funds_sent_to_active_and_retiring_fed_surpasses_locking_cap_before_RSKIP293_no_refund() throws BlockStoreException, BridgeIllegalArgumentException, IOException {
+        BigInteger result = sendFundsSurpassesLockingCapToAnyAddress(
+            false,
+            Coin.FIFTY_COINS,
+            (bridgeConstants, activeFederationAddress, retiringFederationAddress) -> {
+                BtcTransaction tx = new BtcTransaction(bridgeConstants.getBtcParams());
+                tx.addOutput(Coin.FIFTY_COINS.div(2), activeFederationAddress);
+                tx.addOutput(Coin.FIFTY_COINS, retiringFederationAddress);
+                return tx;
+            },
+            true
+        );
+        Assert.assertEquals(co.rsk.core.Coin.fromBitcoin(Coin.FIFTY_COINS.div(2)).asBigInteger(), result);
+    }
+
+    @Test
+    public void registerFastBridgeBtcTransaction_funds_sent_to_active_and_retiring_fed_surpasses_locking_cap_before_RSKIP293_only_funds_sent_to_active_fed_should_be_refunded_to_LP() throws BlockStoreException, BridgeIllegalArgumentException, IOException {
```

Comment:
I don't think this is working as expected.

The method `sendFundsSurpassesLockingCapToAnyAddress` checks that the refunded value is between amountSent and amountSent - fees, right? And amountSent is calculated considering the value sent to both active and retiring federations https://github.com/rsksmart/rskj/blob/reject-funds-for-retiring-fed/rskj-core/src/test/java/co/rsk/peg/BridgeSupportFlyoverTest.java#L791

But we expect to receive a refund only for the amount sent to the active federation, right?

Reply:
Good catch.

This has room for improvement, but even so it is working as expected. When the RSKIP293 isn't active the method `getAmountSentToAddresses` only look for funds sent to one address, the first address in the list. This is the legacy behaviour. So, this is comparing only with the funds sent to the activeFedAddress instead of both fedAddresses.

However, as you I believe this is confusing too. What do you think if I change `getAmountSentToAddresses` method to throw an exception when RSKIP293 isn't active and is being called with a list of two addresses?

![image](https://user-images.githubusercontent.com/94651730/163384334-9b6c6d5b-8bbc-4e00-b088-4f0d4692e87e.png)

Reply:
Anyway, I'm also going to add a condition in `sendFundsSurpassesLockingCapToAnyAddress ` method to only pass the active federation when RSKIP293 isn't active. This will make it clear that this is the only address being used to get the amount sent.

Reply:
I like the idea of changing `sendFundsSurpassesLockingCapToAnyAddress` to get the amount sent only to the active federation if RSKIP 293 is not active. 

I wouldn't throw an exception in `getAmountSentToAddresses` when RSKIP 293 is not active, that was not the behavior pre hop and could cause the network to fork.

Thanks!

Reply:
Perfect. Thanks to you.

Changes pushed.